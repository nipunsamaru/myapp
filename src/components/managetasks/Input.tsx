import {StyleSheet, Text, TextInput, View} from 'react-native';
import React from 'react';
import {Colors} from '../../constants/style';

const Input = ({label, invalid, style, textInputConfig}) => {
  const inputStyles = [styles.input];

  if (textInputConfig && textInputConfig.multiline) {
    inputStyles.push(styles.inputMultiline);
  }
  if (invalid) {
    inputStyles.push(styles.invalidInput);
  }
  return (
    <View style={[styles.inputContainer, style]}>
      <Text style={[styles.label, invalid && styles.invalidLabel]}>
        {label}
      </Text>
      <TextInput style={inputStyles} {...textInputConfig} />
    </View>
  );
};

const styles = StyleSheet.create({
  inputContainer: {
    marginHorizontal: 4,
    marginVertical: 8,
  },
  label: {
    fontSize: 12,
    color: Colors.primary100,
    marginBottom: 4,
  },
  input: {
    backgroundColor: Colors.primary100,
    color: Colors.primary700,
    padding: 6,
    borderRadius: 6,
    fontSize: 18,
  },
  inputMultiline: {
    minHeight: 100,
    textAlignVertical: 'top',
  },
  invalidLabel: {
    color: Colors.error500,
  },
  invalidInput: {
    backgroundColor: Colors.error50,
  },
});

export default Input;
