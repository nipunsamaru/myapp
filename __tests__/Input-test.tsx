import React from 'react';
import {Text, View} from 'react-native';
import renderer from 'react-test-renderer';
import Input from '../src/components/managetasks/Input';
import {Colors} from '../src/constants/style';

describe('Input component', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<Input label="Test Label" />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders label correctly', () => {
    const label = 'Test Label';
    const component = renderer.create(<Input label={label} />);
    const root = component.root;
    const textInstance = root.findByType(Text);
    expect(textInstance.props.children).toEqual(label);
  });

  // Add more test cases as needed
});
