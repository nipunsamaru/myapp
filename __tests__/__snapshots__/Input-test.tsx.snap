// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input component renders correctly 1`] = `
<View
  style={
    [
      {
        "marginHorizontal": 4,
        "marginVertical": 8,
      },
      undefined,
    ]
  }
>
  <Text
    style={
      [
        {
          "color": "#c6affc",
          "fontSize": 12,
          "marginBottom": 4,
        },
        undefined,
      ]
    }
  >
    Test Label
  </Text>
  <TextInput
    style={
      [
        {
          "backgroundColor": "#c6affc",
          "borderRadius": 6,
          "color": "#2d0689",
          "fontSize": 18,
          "padding": 6,
        },
      ]
    }
  />
</View>
`;
